src/noCommentedOutCodeRule.ts

```diff
@@ -42,51 +42,100 @@ class NoCommentedOutCodeRuleWalker extends ErrorTolerantWalker {
     }
 
     private scanCommentForCode(startPosition: number, commentText: string) {
-        if (this.textIsCode(this.cleanComment(commentText))) {
-            this.foundSuspiciousComment(startPosition, commentText);
+        const trimmedCommentText = this.trimTextLines(commentText);
+
+        if (this.isTextCode(trimmedCommentText)) {
+            this.handleSuspiciousComment(startPosition, commentText);
         }
     }
 
-    private textIsCode(text: string): boolean {
-        text = text.trim();
```

Comment:
The removal of `.trim()` causes `isTextSingleWord(text)` to fail.

With a block comment, the input looks like:

```

Lorem

```

While it was `Lorem` before.

Reply:
Fixed in https://github.com/Glavin001/tslint-clean-code/pull/64

Reply:
Shit, you're right. I changed it so spaces (and comment delimiter) will be trimmed from each line separately. The final `.trim()` on the resulting `trimmedLines` after they are joined together is missing, which would remove empty lines at start and end. Sorry for that!

Two suggestions:
* Return `trimmedLines.join('\n').trim()` from `trimTextLines` to remove empty lines at start and end. Maybe even use `trim()` on each line instead of removing whitespace with the RegExp replace.
* Add a test for a single word which is actually commented-out code, like `return` maybe. This way we might be able to make sure, single-word comments are being ignored properly.

Speaking of, what's the intention of this special handling anyway 🌝 Why should a single word always be ignored? Will it always be considered valid code by the algorithm in the linter rule?

Again sorry for the bug!

Reply:
> Again sorry for the bug!

 No worries! Happens to the best of us 😄.


>  what's the intention of this special handling anyway 🌝 Why should a single word always be ignored? Will it always be considered valid code by the algorithm in the linter rule?

Exactly. A single word -- without some special characters -- would always be flagged as code. See https://astexplorer.net/

![image](https://user-images.githubusercontent.com/1885333/47884122-5a424100-de0e-11e8-964f-e6878ffb871a.png)

> Add a test for a single word which is actually commented-out code, like `return` maybe. 

I added more tests in https://github.com/Glavin001/tslint-clean-code/pull/64 specifically for block comments.

Good idea on adding a test for `return`. I did a little investigation and one idea would be:
- Check if single word (using current RegExp)
- Check if parsed AST is single `ExpressionStatement` with an `identifier` equal to the word

See below:

![image](https://user-images.githubusercontent.com/1885333/47884397-506d0d80-de0f-11e8-828d-f8409b8a5a98.png)

It looks like the actual commented-out code with single words would be different type of statements, such as `ReturnStatement` and `FunctionDeclaration`. 🎉 

> Two suggestions:

Feel free to submit another PR if you have the bandwidth 😃 . I appreciated your last PR! Thank you for contributing!